compression/run_length_encoding.py

```diff
@@ -0,0 +1,53 @@
+# https://en.wikipedia.org/wiki/Run-length_encoding
+
+
+def run_length_encode(input: str) -> list:
+    """
+    Performs Run Length Encoding
+    >>> run_length_encode("AAAABBBCCDAA")
+    [('A', 4), ('B', 3), ('C', 2), ('D', 1), ('A', 2)]
+    >>> run_length_encode("A")
+    [('A', 1)]
+    >>> run_length_encode("AA")
+    [('A', 2)]
+    >>> run_length_encode("AAADDDDDDFFFCCCAAVVVV")
+    [('A', 3), ('D', 6), ('F', 3), ('C', 3), ('A', 2), ('V', 4)]
+    """
+    encoded = []
+    count = 1
+
+    for i in range(len(input)):
+        if i + 1 < len(input) and input[i] == input[i + 1]:
+            count += 1
+        else:
+            encoded.append((input[i], count))
+            count = 1
+
+    return encoded
+
+
+def run_length_decode(input: list) -> str:
```

Comment:
https://docs.python.org/3/library/functions.html#input is a builtin function in Python so please pick a different variable name.